src/ConsumeOwnEventsService.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) Martin Da Costa 2023 (martindc.merg@gmail.com)
+// This file is part of VLCB-Arduino project on https://github.com/SvenRosvall/VLCB-Arduino
+// Licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
+// The full licence can be found at: http://creativecommons.org/licenses/by-nc-sa/4.0/
+
+#pragma once
+
+#include "Service.h"
+#include <Controller.h>
+#include <vlcbdefs.hpp>
+
+namespace VLCB {
+
+class Configuration;
+
+class ConsumeOwnEventsService : public Service
+{
+public:
+  virtual void setController(Controller *cntrl) override;
+  virtual Processed handleMessage(unsigned int opc, VlcbMessage *msg) override {return NOT_PROCESSED;}
+  
+  virtual byte getServiceID() override 
+  {
+    return SERVICE_ID_CONSUME_OWN_EVENTS;
+  }
+  virtual byte getServiceVersionID() override 
+  {
+    return 1;
+  }
+  
+  bool available(void);
+  void put(const VlcbMessage *msg);
+  VlcbMessage *peek(void);
+  VlcbMessage *get(void);
+  void clear(void);
+  byte bufUse(void);
+  unsigned int puts();
+  unsigned int gets();
+  unsigned int overflows(void);
+    
+private:
+  Controller *controller;
+  Configuration *module_config;  // Shortcut to reduce indirection code.
+  
+  byte numItems = 4;
```

Comment:
Should we allow users to change the size of the buffer? Add a parameter to the constructor?

Reply:
I thought about this but decided to get some experience on how the buffer behaves.  To that end, I will add a monitor for puts, gets and overflows.  So far, I've only seen the size transition though 1 and back to zero as the buffer gets emptied almost as soon as it has something in it.  If the loop starts to take a long time, it could be a problem.

Reply:
Consider something CANCOMPUTE or CANGATE where you could get many output events when an input changes.

Reply:
Fair point.  I'll leave the default at 4 and add an option to specify something else.